iis/check.py

```diff
@@ -5,183 +5,119 @@
                     continue
-                except TypeError:
-                    self.log.warning(u"When extracting metrics with WMI, found a missing property"
-                                     " '{0}'".format(wmi_property))
-                    continue
-        return metrics
-
-    def _submit_events(self, wmi_sampler, sites):
-        expected_sites = set(sites)
-
-        for wmi_obj in wmi_sampler:
-            sitename = wmi_obj['Name']
-            uptime = wmi_obj["ServiceUptime"]
-            status = AgentCheck.CRITICAL if uptime == 0 else AgentCheck.OK
-
-            self.service_check(self.SERVICE_CHECK, status, tags=['site:{0}'.format(self.normalize(sitename))])
-            expected_sites.remove(sitename)
 
+                for sitename, val in vals.iteritems():
+                    tags = []
+                    if key in self._tags:
+                        tags = self._tags[key]
+
+                    try:
+                        if not counter.is_single_instance():
+                            # Skip any sites we don't specifically want.
+                            if not sites:
+                                tags.append("site:{0}".format(self.normalize(sitename)))
+                            # always report total
+                            elif sitename == "_Total":
+                                tags.append("site:{0}".format(self.normalize(sitename)))
+                            elif sitename not in sites:
+                                continue
+                            else:
+                                tags.append("site:{0}".format(self.normalize(sitename)))
+                    except Exception as e:
+                        self.log.error("Caught exception %s setting tags" % str(e))
+
+                    try:
+                        metric_func(dd_name, val, tags)
+                    except Exception as e:
+                        self.log.error("metric_func: %s %s %s" % (dd_name, str(val), str(e)))
+                        pass
+
+                    if dd_name == "iis.uptime":
+                        uptime = int(val)
+                        status = AgentCheck.CRITICAL if uptime == 0 else AgentCheck.OK
+                        self.service_check(self.SERVICE_CHECK, status, tags=['site:{0}'.format(self.normalize(sitename))])
+                        if sitename in expected_sites:
+                            self.log.debug("Removing %s from expected sites" % sitename)
+                            expected_sites.remove(sitename)
+                        else:
+                            self.log.warning("site not in expected_sites %s" % sitename)
+
+            except Exception as e:
+                # don't give up on all of the metrics because one failed
+                self.log.error("IIS Failed to get data for %s %s: %s" % (inst_name, dd_name, str(e)))
+                pass
+        self.log.debug("expected sites is now %s" % str(expected_sites))
```

Comment:
Can you change this to make it a bit more informative? Something like "failed to grab metrics for...", so that we don't have to know the internal code of the check to interpret it. 
Additional questions: 
Do we want to log this on info?
Do we want to log it on warning, so it shows up on the info page?